log4j-core/src/main/java/org/apache/logging/log4j/core/appender/mom/kafka/KafkaManager.java

```diff
@@ -73,17 +74,19 @@ public boolean releaseSub(final long timeout, final TimeUnit timeUnit) {
     private void closeProducer(final long timeout, final TimeUnit timeUnit) {
         if (producer != null) {
             // This thread is a workaround for this Kafka issue: https://issues.apache.org/jira/browse/KAFKA-1660
-            final Runnable task = new Runnable() {
+           final Thread closeThread = new Log4jThread(new Runnable() {
                 @Override
                 public void run() {
                     if (producer != null) {
                         producer.close();
                     }
                 }
-            };
+            }, "KafkaManager-CloseThread");
+            closeThread.setDaemon(true); // avoid blocking JVM shutdown
+            closeThread.start();
             try {
-                getLoggerContext().submitDaemon(task).get(timeout, timeUnit);
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                closeThread.join(timeUnit.toMillis(timeout));
+            } catch (final InterruptedException ignore) {
```

Comment:
The thread would have been interrupted for a reason, so it's generally good to handle the exception and reset the interrupt flag (it doesn't necessarily cause an exception to be thrown; code has to manually check a thread's interrupt status according to the javadocs).